src/java/org/apache/cassandra/guardrails/Guardrails.java

```diff
@@ -41,203 +36,214 @@
+                          () -> config.tombstone_warn_threshold,
+                          () -> config.tombstone_failure_threshold,
+                          (isWarning, what, v, t) -> isWarning ?
+                                                     format("Scanned over %s tombstone rows for query %1.512s - more than the warning threshold %s", v, what, t) :
+                                                     format("Scanned over %s tombstone rows during query %1.512s - more than the maximum allowed %s; query aborted", v, what, t));
 
 
     public static final Threshold batchSize =
-    new SizeThreshold("batch_size",
-                      config::getBatchSizeWarnThreshold,
-                      config::getBatchSizeFailThreshold,
-                      (isWarning, what, v, t) -> isWarning
-                                                 ? format("Batch for %s is of size %s, exceeding specified warning threshold %s", what, v, t)
-                                                 : format("Batch for %s is of size %s, exceeding specified failure threshold %s", what, v, t));
+            factory.threshold("batch_size",
+                              config::getBatchSizeWarnThreshold,
+                              config::getBatchSizeFailThreshold,
+                              (isWarning, what, v, t) -> isWarning
+                                                ? format("Batch for %s is of size %s, exceeding specified warning threshold %s", what, formatSize(v), formatSize(t))
+                                                : format("Batch for %s is of size %s, exceeding specified failure threshold %s", what, formatSize(v), formatSize(t)));
 
     public static final Threshold unloggedBatchAcrossPartitions =
-    new Threshold("unlogged_batch_across_partitions",
-                  () -> config.unlogged_batch_across_partitions_warn_threshold,
-                  () -> -1L,
-                  (x, what, v, t) -> format("Unlogged batch covering %s partitions detected " +
-                                            "against table%s %s. You should use a logged batch for " +
-                                            "atomicity, or asynchronous writes for performance.",
-                                            v, what.contains(", ") ? "s" : "", what));
+            factory.threshold("unlogged_batch_across_partitions",
+                          () -> config.unlogged_batch_across_partitions_warn_threshold,
+                          () -> -1L,
+                          (x, what, v, t) -> format("Unlogged batch covering %s partitions detected " +
+                                                    "against table%s %s. You should use a logged batch for " +
+                                                    "atomicity, or asynchronous writes for performance.",
+                                                    v, what.contains(", ") ? "s" : "", what));
+
+    private static String formatSize(long size)
```

Comment:
nit: maybe name `formatSizeBytes` or similar to more clearly indicate the size units?